src/creditly.js

```diff
@@ -110,11 +127,15 @@ var Creditly = (function() {
         removeErrors(element);
         var number = getInputValue(e, selector);
         var isAmericanExpressCard = isAmericanExpress(number);
-        var maximumLength = getMaximumLength(isAmericanExpressCard);
+        var isDinersClubCard = isDinersClub(number);
+        var cardBrandName = cardBrand(number);
+        var maximumLength = getMaximumLength(cardBrandName);
         if (shouldProcessInput(e, maximumLength, selector)) {
           var newInput;
           if (isAmericanExpressCard) {
             newInput = addSpaces(number, americanExpressSpaces);
+          } else if (isDinersClubCard) {
```

Comment:
@wangjohn 
OK! I just fixed it. this like https://github.com/kurotaky/creditly/commit/1a70c243adeaf4d057449bfe61b53bbe371a621d.